db/migrations/mysql/000006_create_incoming_webhooks.up.sql

```diff
@@ -0,0 +1,177 @@
+CREATE TABLE IF NOT EXISTS IncomingWebhooks (
+    Id varchar(26) NOT NULL,
+    CreateAt bigint(20) DEFAULT NULL,
+    UpdateAt bigint(20) DEFAULT NULL,
+    DeleteAt bigint(20) DEFAULT NULL,
+    UserId varchar(26) DEFAULT NULL,
+    ChannelId varchar(26) DEFAULT NULL,
+    TeamId varchar(26) DEFAULT NULL,
+    DisplayName varchar(64) DEFAULT NULL,
+    Description text,
+    Username varchar(255) DEFAULT NULL,
+    IconURL text,
+    PRIMARY KEY(Id)
+);
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'IncomingWebhooks'
+        AND table_schema = DATABASE()
+        AND column_name = 'UserName'
+    ) > 0,
+    'SELECT 1',
+    'ALTER TABLE IncomingWebhooks ADD UserName text;'
+));
+
+PREPARE alterIfNotExists FROM @preparedStatement;
+EXECUTE alterIfNotExists;
+DEALLOCATE PREPARE alterIfNotExists;
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'IncomingWebhooks'
+        AND table_schema = DATABASE()
+        AND column_name = 'IconURL'
+    ) > 0,
+    'SELECT 1',
+    'ALTER TABLE IncomingWebhooks ADD IconURL text;'
+));
+
+PREPARE alterIfNotExists FROM @preparedStatement;
+EXECUTE alterIfNotExists;
+DEALLOCATE PREPARE alterIfNotExists;
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'IncomingWebhooks'
+        AND table_schema = DATABASE()
+        AND column_name = 'ChannelLocked'
+    ) > 0,
+    'SELECT 1',
+    'ALTER TABLE IncomingWebhooks ADD ChannelLocked tinyint(1) DEFAULT 0;'
+));
+
+PREPARE alterIfNotExists FROM @preparedStatement;
+EXECUTE alterIfNotExists;
+DEALLOCATE PREPARE alterIfNotExists;
+
+SET @preparedStatement = (SELECT IF(
+    (
+        SELECT COUNT(*) FROM INFORMATION_SCHEMA.COLUMNS
+        WHERE table_name = 'IncomingWebhooks'
+        AND table_schema = DATABASE()
+        AND column_name = 'Description'
+    ) > 0,
+    'SELECT 1',
+    'ALTER TABLE IncomingWebhooks ADD Description text;'
+));
+
+PREPARE alterIfNotExists FROM @preparedStatement;
+EXECUTE alterIfNotExists;
+DEALLOCATE PREPARE alterIfNotExists;
```

Comment:
I can't find an ALTER in `upgrade.go` that adds this column, but I've found this one that changes its type:

```go
sqlStore.AlterColumnTypeIfExists("IncomingWebhooks", "Description", "varchar(500)", "varchar(500)")
```

Shouldn't we be changing the type here instead?

Reply:
100% copy + 🍝 error